react-app/src/App.js

```diff
@@ -0,0 +1,115 @@
+import './App.css';
+import React, {useState} from "react";
+
+import AddButton from "./addButton";
+import DeleteButton from "./deleteButton";
+import TaskList from './taskList';
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import Alert from "./Alert";
+
+function App(props) {
+
+    const [todo, setTodo] = useState(props.initialData);
+
+    const [completedList, setCompletedList] = useState(todo.filter(item => item.isChecked));
+    const [unCompletedList, setUnCompletedList] = useState(todo.filter(item => !item.isChecked));
+
+    const [showAlert, setShowAlert] = useState(false);
+
+    function handleItemChanged(itemId, field, newValue) {
+        const newData1 = completedList.map(
+            p => p.id === itemId ? {...p, [field]: newValue} : p)
+        setCompletedList(newData1)
+        const newData2 = unCompletedList.map(
+            p => p.id === itemId ? {...p, [field]: newValue} : p)
+        setUnCompletedList(newData2)
+    }
+
+    function moveTasks(checked, item) {
+        if (completedList.includes(item)) {
+            setUnCompletedList([...unCompletedList,
+                    {
+                        id: item.id,
+                        isChecked: false,
+                        textInput: item.textInput,
+                        isBlur:false
+                    }]);
+            setCompletedList(completedList.filter(element => element.id !== item.id))
+
+        } else {
+            setCompletedList([...completedList,
+                {
+                    id: item.id,
+                    isChecked: true,
+                    textInput: item.textInput,
+                    isBlur: true
+                }]);
+            setUnCompletedList(unCompletedList.filter(element => element.id !== item.id));
+        }
+    }
+
+    function handleCompletedDeleted() {
+        setCompletedList([]);
+    }
+
+    function handleItemAdded() {
+        setUnCompletedList([...unCompletedList,
+            {
+                id: generateUniqueID(),
+                isChecked: false,
+                textInput: "",
+                isBlur: false
+            }])
+    }
+
+    function alertDelete() {
+        console.log("pressed delete")
+        handleCompletedDeleted()
+    }
+
+    function toggleModal() {
+        setShowAlert(!showAlert);
+    }
+
+    return <>
+        <h1>To Do List</h1>
+        <TaskList
+            todo={todo}
+            completedList={completedList}
```

Comment:
I would just pass todo={unCompletedList} or todo={completedList}  and then don't bother with passing in completedList or unCompletedList attributes (or setUncompletedList or setCompletedList).  onItemChanged should be sufficient. (also no need for moveTasks).